internal/services/qumulo/qumulo_file_system_resource.go

```diff
@@ -0,0 +1,274 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+var _ sdk.Resource = FileSystemResource{}
+var _ sdk.ResourceWithUpdate = FileSystemResource{}
+
+const (
+	offerId     = "qumulo-saas-mpp"
+	publisherId = "qumulo1584033880660"
+)
+
+type FileSystemResource struct{}
+
+func (r FileSystemResource) ModelObject() interface{} {
+	return &FileSystemResourceSchema{}
+}
+
+type FileSystemResourceSchema struct {
+	AvailabilityZone  string                 `tfschema:"availability_zone"`
+	AdminPassword     string                 `tfschema:"admin_password"`
+	DelegatedSubnetId string                 `tfschema:"delegated_subnet_id"`
+	InitialCapacity   int64                  `tfschema:"initial_capacity"`
+	Location          string                 `tfschema:"location"`
+	MarketplacePlanId string                 `tfschema:"marketplace_plan_id"`
+	Name              string                 `tfschema:"name"`
+	ResourceGroupName string                 `tfschema:"resource_group_name"`
+	StorageSku        string                 `tfschema:"storage_sku"`
+	Tags              map[string]interface{} `tfschema:"tags"`
+	UserEmailAddress  string                 `tfschema:"user_email_address"`
+}
+
+func (r FileSystemResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return filesystems.ValidateFileSystemID
+}
+
+func (r FileSystemResource) ResourceType() string {
+	return "azurerm_qumulo_file_system"
+}
+
+func (r FileSystemResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			ForceNew:     true,
+			Required:     true,
+			Type:         pluginsdk.TypeString,
+			ValidateFunc: validate.FileSystemName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"admin_password": {
+			Type:      pluginsdk.TypeString,
+			Required:  true,
+			ForceNew:  true,
+			Sensitive: true,
+		},
+
+		"delegated_subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"initial_capacity": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.IntBetween(18, 1000),
+		},
+
+		"marketplace_plan_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"storage_sku": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringInSlice(filesystems.PossibleValuesForStorageSku(), false),
+		},
+
+		"user_email_address": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"availability_zone": commonschema.ZoneSingleOptionalForceNew(),
```

Comment:
[In 3.0](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-upgrade-guide#behavioural-updates) the inconsistent naming for Zones has been fixed into `zone` (for a single zone) and `zones` (for multiple zones) for consistency/with consistent behaviours - can we fix the naming here?